Error in the end of the `make install`:

```
Traceback (most recent call last):
  File "${src}/Lib/runpy.py", line 163, in _run_module_as_main
    mod_name, _Error)
  File "${src}/Lib/runpy.py", line 111, in _get_module_details
    __import__(mod_name)  # Do not catch exceptions initializing package
  File "${src}/Lib/ensurepip/__init__.py", line 9, in <module>
    import tempfile
  File "${src}/Lib/tempfile.py", line 35, in <module>
    from random import Random as _Random
  File "${src}/Lib/random.py", line 885, in <module>
    _inst = Random()
  File "${src}/Lib/random.py", line 97, in __init__
    self.seed(x)
  File "${src}/Lib/random.py", line 113, in seed
    a = long(_hexlify(_urandom(2500)), 16)
OSError: [Errno 38] Function not implemented
make: *** [Makefile:927: install] Error 1
```

While the normal installing process is following:

```
Collecting setuptools
Collecting pip
Installing collected packages: setuptools, pip
Successfully installed pip-9.0.1 setuptools-28.8.0
```

Discussion found here: http://bugs.python.org/issue29188
Patch found in the discussion above:
https://hg.python.org/cpython/rev/13a39142c047

Chunk #1 is removed from the original patch because it rejected.


# HG changeset patch
# User Victor Stinner <victor.stinner@gmail.com>
# Date 1483956641 -3600
# Node ID 13a39142c0473ecb64fcd4b12a915025df6e4310
# Parent  cb4f73be9486d47f1dc4285998d1532d8857c59e
Don't use getentropy() on Linux

Issue #29188: Support glibc 2.24 on Linux: don't use getentropy() function but
read from /dev/urandom to get random bytes, for example in os.urandom().  On
Linux, getentropy() is implemented which getrandom() is blocking mode, whereas
os.urandom() should not block.

diff --git a/Python/random.c b/Python/random.c
--- a/Python/random.c
+++ b/Python/random.c
@@ -97,8 +97,15 @@ win32_urandom(unsigned char *buffer, Py_
 }
 
 /* Issue #25003: Don't use getentropy() on Solaris (available since
- * Solaris 11.3), it is blocking whereas os.urandom() should not block. */
-#elif defined(HAVE_GETENTROPY) && !defined(sun)
+   Solaris 11.3), it is blocking whereas os.urandom() should not block.
+
+   Issue #29188: Don't use getentropy() on Linux since the glibc 2.24
+   implements it with the getrandom() syscall which can fail with ENOSYS,
+   and this error is not supported in py_getentropy() and getrandom() is called
+   with flags=0 which blocks until system urandom is initialized, which is not
+   the desired behaviour to seed the Python hash secret nor for os.urandom():
+   see the PEP 524 which was only implemented in Python 3.6. */
+#elif defined(HAVE_GETENTROPY) && !defined(sun) && !defined(linux)
 #define PY_GETENTROPY 1
 
 /* Fill buffer with size pseudo-random bytes generated by getentropy().

